### PR TITLE
Add option to use data (prescribed) ice-shelf melt fluxes in MPAS-Ocean

### PR DESCRIPTION
**Pull request by xylar**
This merge adds a new package and mode for data melt fluxes. When `config_land_ice_flux_mode = 'data'`, land-ice freshwater and sensible-heat fluxes are assigned from the prescribed fields, which can be read in at initialization or from a stream.

### CODE REVIEW

**Comment by xylar** [comment]
This branch can be used in conjunction with https://github.com/MPAS-Dev/compass/pull/580 to create the necessary file for prescribing melt fluxes for a given mesh.

I need to stage the appropriate file for the existing ECwISC30to60E2r1 mesh so we can test this as a stealth feature.

**Comment by xylar** [comment]
@jonbob, @darincomeau and @cbegeman, I could use your help figuring out a good way to turn on prescribed ISMF in E3SM compsets.  Here's the current code:
https://github.com/E3SM-Ocean-Discussion/E3SM/blob/c1bd30de22a9bd000315576e2552a0735ef84040/components/mpas-ocean/cime_config/config_component.xml#L44-L49
One thought I had was `PISMF`, but that makes it hard to detect `PISMF` vs. `ISMF` in the compset name.  Should we add something to `ISMF` (like `IISMF` for interactive ISMF)?  Or should we go with `PSMF` ("prescribed shelf melt fluxes") to keep things simpler?

Once we decide on this, I'll add the appropriate check in the file above and also add the 8 compsets in:
https://github.com/E3SM-Ocean-Discussion/E3SM/blob/c1bd30de22a9bd000315576e2552a0735ef84040/components/mpas-ocean/cime_config/config_compsets.xml
and the 3 in:
https://github.com/E3SM-Project/E3SM/blob/4deb2611a4293fdb578db5dd1ba9fd7a6c223029/cime_config/allactive/config_compsets.xml#L290-L303

Presumably, we won't add anything else in this PR.

**Comment by cbegeman** [comment]
> One thought I had was PISMF, but that makes it hard to detect PISMF vs. ISMF in the compset name. Should we add something to ISMF (like IISMF for interactive ISMF)? Or should we go with PSMF ("prescribed shelf melt fluxes") to keep things simpler?

@xylar I don't have a strong preference between `IISMF` vs. `PISMF` and `ISMF` vs. `PSMF`. Maybe a slight preference for the former.

**Comment by darincomeau** [comment]
@xylar back when we added the data iceberg forcing, we used "D"ata and "P"rognostic icebergs, `DIB` vs. `PIB`.
https://github.com/E3SM-Project/E3SM/blob/master/components/mpas-seaice/cime_config/config_component.xml#L29

Perhaps that was poor choice, as we've all seemed to migrate to "prescribed" over "data" forcing. But I don't think we want `P` meaning one thing for icebergs, and the opposite for ISMF.

1. Stick with consistency and use `DISMF` here?
2. Go back and change `DIB` to `PIB`? But that would involve changing the compset names with `DIB` we've been using for years...

Not sure what the best solution is, open to whatever you think best.

**Comment by xylar** [comment]
@darincomeau, thanks.  My sense, then, is that we should go with `DISMF` (data ) and `PISMF` (prognostic).

@jonbob, would the change from `ISMF` to `PISMF` cause trouble?  Are there rules against renaming compsets in E3SM?

I guess it will be too confusing to have `config_land_ice_flux_mode = 'prescribed'` if `PISMF` has a P for prognostic.  Should it be `'data'`?  I'm not super happy with that but it feels like the more inconsistencies between E3SM and MPAS-Ocean's internal naming convention (e.g. `internal` vs. `standalone`), the more confusing things get for developers and users.

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml [hunk 1]
```diff
@@ -333,12 +333,12 @@
 
 <!-- land_ice_fluxes -->
```
Comment (cbegeman) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:334]:
```suggestion
<!-- land_ice_fluxes -->
<!-- These values of `config_land_ice_flux_mode` will be used when `ocn_ismf` is "none" -->
```

**Review by cbegeman — APPROVED** [review]
@xylar This looks correct to me. Let me know if you'd like help with testing.

**Comment by cbegeman** [comment]
@xylar Just a heads up that you will want to make similar changes to accommodate the new `config_land_ice_flux_mode` option https://github.com/E3SM-Project/E3SM/pull/5464#issuecomment-1491091874

**Comment by xylar** [comment]
Thanks for the tip, @cbegeman.  I have included the mct driver changes here, too.

File: components/mpas-ocean/driver/ocn_comp_mct.F [hunk 2]
```diff
@@ -792,6 +792,7 @@ end subroutine xml_stream_get_attributes
     call mpas_pool_get_config(domain % configs, 'config_land_ice_flux_mode', config_land_ice_flux_mode)
     if ( trim(config_land_ice_flux_mode) == 'off' .or. &
          trim(config_land_ice_flux_mode) == 'pressure_only' .or. &
+         trim(config_land_ice_flux_mode) == 'prescribed' .or. &
```
Comment (xylar) [inline components/mpas-ocean/driver/ocn_comp_mct.F:795]: Amid the withespace changes, this is the only "real change".

**Comment by xylar** [comment]
I have rebased this and the compass branch.  Next, I will re-generate and stage the appropriate data melt flux file for the existing ECwISC mesh for testing.  Then, I will see how a 5 day (then a 10 year) simulation in E3SM goes.

**Comment by xylar** [comment]
Progress is being hampered by https://github.com/E3SM-Project/E3SM/issues/5598, but I can roll back where I base my branch from to avoid that issue for now.

**Comment by xylar** [comment]
I ran a 5-year G-case with this branch.  Results are here and look good to me:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xasay-davis/analysis/20230515.GMPAS-IAF-DISMF.T62_ECwISC30to60E2r1.chrysalis/

File: components/mpas-ocean/cime_config/buildnml [hunk 3]
```diff
@@ -277,8 +278,8 @@ def buildnml(case, caseroot, compname):
         if ocn_ic_mode == 'spunup':
             ic_date = '230220'
             ic_prefix = 'mpaso.ECwISC30to60E2r1.rstFromG-anvil'
-
-
+        if ocn_ismf == 'data':
+            data_ismf_file = 'prescribed_ismf_adusumilli2020.ECwISC30to60E2r1.230429.nc'
```
Comment (xylar) [inline components/mpas-ocean/cime_config/buildnml:282]: I need to add this data file for SORRM.  I think that's the only other mesh we need to support for now.

**Comment by xylar** [comment]
Here are the analysis results for a 5-year SORRM G-case with data melt rates:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xasay-davis/analysis/20230516.GMPAS-IAF-DISMF.T62_SOwISC12to60E2r4.chrysalis/

Everything looks good and the melt rates averaged over regions on the SORRM mesh are more similar to the averages over the same regions from the original Adusumilli data than on the EC mesh (as expected because of the higher resolution).  Even so, we consistently slightly underestimate the melt, presumably because of missing portions of the floating ice domain and/or interpolation artifacts.

**Comment by xylar** [comment]
This is a stealth feature so I need to add a new test.

**Comment by xylar** [comment]
It looks like we want the stealth test to run on the `oQU240wLI` mesh, so I'll need to create the data ISMF for that mesh.

**Comment by xylar** [comment]
I actually don't think this counts as a stealth feature because you can set up and run both E3SM tests and cases right out of the box with `DISMF`.  So I will add some extended tests for it following what we have for PISMF.

**Comment by xylar** [comment]
Closed in favor of https://github.com/E3SM-Project/E3SM/pull/5698

**Pull request closed without merging**